netboot/netboot.go

```diff
@@ -46,6 +47,37 @@ func RequestNetbootv6(ifname string, timeout time.Duration, retries int, modifie
 	return conversation, nil
 }
 
+// RequestNetbootv4 sends a netboot request via DHCPv4 and returns the exchanged packets. Additional modifiers
+// can be passed to manipulate both the discover and offer packets.
+func RequestNetbootv4(ifname string, timeout time.Duration, retries int, modifiers ...dhcpv4.Modifier) ([]*dhcpv4.DHCPv4, error) {
+	var (
+		conversation []*dhcpv4.DHCPv4
+		err          error
+	)
+	delay := 2 * time.Second
+	modifiers = append(modifiers, dhcpv4.WithNetboot)
+	for i := 0; i <= retries; i++ {
+		log.Printf("sending request, attempt #%d", i+1)
+		client := dhcpv4.NewClient()
+		client.ReadTimeout = timeout
+		conversation, err = client.Exchange(ifname, nil, modifiers...)
+		if err != nil {
+			log.Printf("Client.Exchange failed: %v", err)
+			log.Printf("sleeping %v before retrying", delay)
+			if i >= retries {
+				// don't wait at the end of the last attempt
+				break
+			}
+			time.Sleep(delay)
```

Comment:
calls that sleep are not test-friendly - what about making the initial sleep time configurable, or the sleep function configurable? They could be top-level variables so that the caller can override them (e.g. `netboot.Sleeper` and `netboot.InitialRetryInterval`). Another option for the interval is to become a function argument, but I'd rather go for top-level var to conform with the v6 version of this function.

Reply:
Not sure what you mean with "conform with the v6 version of this function", as they are making identical use of sleep. I will amend both accordingly. Change in function arguments breaks the API if I want to align `RequestNetbootv6`, so I don't think it's an option.

Reply:
On the interface, I was talking about the interface, not the content of the function.
Thanks for fixing the v6 version too!